app/src/main/java/org/oppia/android/app/onboarding/CreateProfileFragmentPresenter.kt

```diff
@@ -107,19 +142,102 @@ class CreateProfileFragmentPresenter @Inject constructor(
 
     binding.onboardingNavigationBack.setOnClickListener { activity.finish() }
     binding.createProfileEditPictureIcon.setOnClickListener {
-      activityResultLauncher.launch(
-        galleryIntent
-      )
+      activityResultLauncher.launch(galleryIntent)
     }
     binding.createProfilePicturePrompt.setOnClickListener {
-      activityResultLauncher.launch(
-        galleryIntent
-      )
+      activityResultLauncher.launch(galleryIntent)
     }
     binding.createProfileUserImageView.setOnClickListener {
-      activityResultLauncher.launch(
-        galleryIntent
-      )
+      activityResultLauncher.launch(galleryIntent)
     }
   }
+
+  private fun updateProfileDetails(profileName: String) {
+    profileManagementController.updateNewProfileDetails(
+      profileId = profileId,
+      profileType = profileType,
+      avatarImagePath = selectedImageUri,
+      colorRgb = selectUniqueRandomColor(),
+      newName = profileName,
+      isAdmin = true
+    ).toLiveData().observe(
+      fragment,
+      { result ->
+        when (result) {
+          is AsyncResult.Success -> {
+            createProfileViewModel.hasErrorMessage.set(false)
+
+            val params = IntroActivityParams.newBuilder()
+              .setProfileNickname(profileName)
+              .build()
+
+            val intent =
+              IntroActivity.createIntroActivity(activity).apply {
+                putProtoExtra(IntroActivity.PARAMS_KEY, params)
+                decorateWithUserProfileId(profileId)
+              }
+
+            fragment.startActivity(intent)
+          }
+          is AsyncResult.Failure -> {
+            createProfileViewModel.hasErrorMessage.set(true)
+
+            val errorMessage = when (result.error) {
+              is ProfileManagementController.ProfileNameOnlyLettersException ->
+                appLanguageResourceHandler.getStringInLocale(
+                  R.string.add_profile_error_name_only_letters
+                )
+              else -> result.error.localizedMessage
+            }
+
+            createProfileViewModel.errorMessage.set(errorMessage)
+
+            oppiaLogger.e(
+              "CreateProfileFragment",
+              "Failed to update profile details.",
+              result.error
+            )
+          }
+          is AsyncResult.Pending -> {}
+        }
+      }
+    )
+  }
+
+  /** Randomly selects a color for the new profile that is not already in use. */
+  private fun selectUniqueRandomColor(): Int {
+    return COLORS_LIST.map {
+      ContextCompat.getColor(fragment.requireContext(), it)
+    }.random()
```

Comment:
```suggestion
    return ContextCompat.getColor(fragment.requireContext(), COLORS_LIST.random())
```

Bit more compact and efficient.

Reply:
Updated